pkgs/by-name/li/little_boxes/package.nix

```diff
@@ -0,0 +1,57 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, installShellFiles
+, testers
+, little_boxes
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "little_boxes";
```

Comment:
```suggestion
  pname = "little-boxes";
```

See https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming:

> Dashes in the package pname should be preserved in new variable names, rather than converted to underscores or camel cased — e.g., http-parser instead of http_parser or httpParser. The hyphenated style is preferred in all three package names.

Also for the attribute name (`by-name` directory name) I think.

Reply:
The block of text you highlighted does not say "underscores **should** be converted to dashes". It says dashes **should not** be converted to underscores.

Further, the whole text points to avoid changing the name for `pname`.

Further, as far as I remember, SDL libs are underscored.

Reply:
> Further, the whole text points to avoid changing the name for pname.

**The hyphenated style is preferred in all three package names**.

> Further, as far as I remember, SDL libs are underscored:

That does not prove that this preference is correct, or that it is correct to always adopt a package name the same with the upstream in this regard.

Reply:
> The hyphenated style is preferred in all three package names.

"preferred" does not imply "mandatory"; especially it is not blocking.

> That does not prove that this preference is correct

The burden of proving it is not correct and should/must be fixed (whether here or in SDL_xxxx) is yours.

And the block of text you provided falls short on sustaining this burden.

> , or that it is correct to always adopt a package name the same with the upstream in this regard.

The burden of proving it is not correct and should/must be fixed (whether here or in SDL_xxxx) is yours.

Also, pulling the docs:

> The pname attribute should be identical to the upstream package name.

> The key words must, must not, required, shall, shall not, should, should not, recommended, may, and optional in this section are to be interpreted as described in RFC 2119. Only emphasized words are to be interpreted in this way.

> This word [must], or the adjective "RECOMMENDED", mean that there may exist valid reasons in particular circumstances to ignore a particular item, but the full implications must be understood and carefully weighed before choosing a different course.

---------

At best this is nitpick, and it resides on the sole discretion of package writer.

Otherwise, if you want a different rewording, please open a pull request against the docs. The purpose of the current docs is clearly not clear enough for you.

Reply:
As long as you don't care, I don't care either